dask_expr/_expr.py

```diff
@@ -54,6 +54,18 @@ def __init__(self, *args, **kwargs):
                 operands.append(type(self)._defaults[parameter])
         assert not kwargs
         self.operands = operands
+        if self._required_attribute:
+            dep = next(iter(self.dependencies()))._meta
+            if not hasattr(dep, self._required_attribute):
+                # Raise a ValueError instead of AttributeError to
+                # avoid infinite recursion
+                raise ValueError(f"{dep} has no attribute {self._required_attribute}")
+
+    @property
+    def _required_attribute(self) -> str:
+        # Specify if the first `dependency` must support
+        # a specific attribute for valid behavior.
+        return None
```

Comment:
@phofl - I ran into quite a few cases where the cudf backend was hanging (rather than quickly failing) for cases where a `cudf.DataFrame`/`Series` did not support the same attribute as `pd.DataFrame`/`Series` (e.g. `nbytes`, `align`, etc).

What do you think about doing something like this so that we can more-quickly detect that the dependency is missing a necessary attribute?

Reply:
Oh yes I like this.